docs/articles_en/learn-openvino/llm_inference_guide/genai-guide/genai-use-cases.rst

```diff
@@ -0,0 +1,305 @@
+               #include "openvino/genai/text2image/pipeline.hpp"
+
+               #include "imwrite.hpp"
+
+               int32_t main(int32_t argc, char* argv[]) {
+                   OPENVINO_ASSERT(argc >= 3 && (argc - 3) % 2 == 0, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>' [<LORA_SAFETENSORS> <ALPHA> ...]]");
+
+                   const std::string models_path = argv[1], prompt = argv[2];
+                   const std::string device = "CPU";  // GPU, NPU can be used as well.
+
+                   ov::genai::AdapterConfig adapter_config;
+                   // Applying Multiple LoRA adapters simultaneously is supported. Parse them all and the corresponding alphas from cmd parameters:
+                   for(size_t i = 0; i < (argc - 3)/2; ++i) {
+                       ov::genai::Adapter adapter(argv[3 + 2*i]);
+                       float alpha = std::atof(argv[3 + 2*i + 1]);
+                       adapter_config.add(adapter, alpha);
+                   }
+
+                   // LoRA adapters passed to the constructor will be activated by default in the next generation.
+                   ov::genai::Text2ImagePipeline pipe(models_path, device, ov::genai::adapters(adapter_config));
+
+                   std::cout << "Generating image with LoRA adapters applied, resulting image will be in lora.bmp\n";
+                   ov::Tensor image = pipe.generate(prompt,
+                       ov::genai::random_generator(std::make_shared<ov::genai::CppStdGenerator>(42)),
+                       ov::genai::width(512),
+                       ov::genai::height(896),
+                       ov::genai::num_inference_steps(20));
+                   imwrite("lora.bmp", image, true);
+
+                   std::cout << "Generating image without LoRA adapters applied, resulting image will be in baseline.bmp\n";
+                   image = pipe.generate(prompt,
+                       ov::genai::adapters(),  // Passing adapters as generation overrides set in the constructor; adapters() means no adapters.
+                       ov::genai::random_generator(std::make_shared<ov::genai::CppStdGenerator>(42)),
+                       ov::genai::width(512),
+                       ov::genai::height(896),
+                       ov::genai::num_inference_steps(20));
+                   imwrite("baseline.bmp", image, true);
+
+                   return EXIT_SUCCESS;
+               }
+
+
+      For more information, refer to the
+      `C++ sample <https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/cpp/text2image/README.md>`__
+
+Using GenAI in Speech Recognition
+#################################
+
+
+The application, shown in code samples below, performs inference on speech
+recognition Whisper Models. The samples include the ``WhisperPipeline`` class
+and use audio files in WAV format at a sampling rate of 16 kHz as input.
+
+.. tab-set::
+
+   .. tab-item:: Python
+      :sync: cpp
+
+      .. code-block:: python
+
+         import argparse
```

Comment:
I would suggest using a more universal code here and in other examples that can be used both in python and from the command line, for example:

``` python
import argparse
import openvino_genai
import librosa


def read_wav(filepath):
    raw_speech, samplerate = librosa.load(filepath, sr=16000)
    return raw_speech.tolist()


def infer(model_dir: str, wav_file_path: str):
    raw_speech = read_wav(wav_file_path)
    pipe = openvino_genai.WhisperPipeline(model_dir)

    def streamer(word: str) -> bool:
        print(word, end="")
        return False

    result = pipe.generate(
        raw_speech,
        max_new_tokens=100,
        # The 'task' and 'language' parameters are supported for multilingual models only.
        language="<|en|>",
        task="transcribe",
        return_timestamps=True,
        streamer=streamer,
    )

    print()
    for chunk in result.chunks:
        print(f"timestamps: [{chunk.start_ts}, {chunk.end_ts}] text: {chunk.text}")
    

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("model_dir")
    parser.add_argument("wav_file_path")
    args = parser.parse_args()
    
    infer(args.model_dir, args.wav_file_path)


if __name__ == '__main__':
    main()
```
 or just 

```python 
import openvino_genai
import librosa


def read_wav(filepath):
    raw_speech, samplerate = librosa.load(filepath, sr=16000)
    return raw_speech.tolist()


def infer(model_dir: str, wav_file_path: str):
    raw_speech = read_wav(wav_file_path)
    pipe = openvino_genai.WhisperPipeline(model_dir)

    def streamer(word: str) -> bool:
        print(word, end="")
        return False

    result = pipe.generate(
        raw_speech,
        max_new_tokens=100,
        # The 'task' and 'language' parameters are supported for multilingual models only.
        language="<|en|>",
        task="transcribe",
        return_timestamps=True,
        streamer=streamer,
    )

    print()
    for chunk in result.chunks:
        print(f"timestamps: [{chunk.start_ts}, {chunk.end_ts}] text: {chunk.text}")
```

This will help you reuse the example code in different scenarios without having to rewrite it.

Reply:
@andrei-kochin, @Wovchena, @slyalin How about the suggestions above? Do we include them to the documentation? If so, which is preferrable?

Reply:
@sgolebiewski-intel  I vote for the option 2 (without the main)